rfcs/convergence/exposing-popper-options.md

```diff
@@ -0,0 +1,184 @@
+  // also can spread calloutProps
+  calloutMaxHeight
+  bounds
+  directionalHintFixed
+/>
+```
+
+v0:
+
+```tsx
+<Popup
+  align
+  position
+  offset
+/>
+
+<Menu
+  items={[
+    { menu: popper: { align, position, offset }}
+  ]}
+/>
+```
+
+## Problem statement
+
+There is a subset of `usePopper` options that are currently used by all converged components that need positioning.
+These options are spread into the component's props:
+
+```tsx
+<Tooltip position="bottom" align="center" />
+<Popover position="bottom" align="center" />
+<Menu position="bottom" align="center" />
+```
+
+There are other popper options that are used in some components but not others, examples of these options are:
+
+- `coverTarget` - allows the positioned element to cover the target
+
+There are also additional options that are not used by any converge component that might be used in the future like:
+
+- `autoSize` - Allows the positioned element to change size based on viewport size
+- `flipBoundary` - Determines what bounds to use to flip the positioning
+
+Issues and bugs can be triggered because certain use cases require more precise configuration of popper. The most likely
+result would be that a popper configuration is exposed for one component but not others.
+
+## Detailed Design or Proposal
+
+### Equal popper configuration
+
+Each converged component that uses popper should allow an equal level of configuration. This ensures that our positioned
+components can behave consistently across the library. Any fixes that require exposing another popper configuration will
+be available to **all** components that use popper.
+
+This proposal does not mean that all popper options must be public, but that if one component exposes a popper configuration,
+then all components should do so equally.
+
+We can achieve this with an agreement that the current `PositioningProps` would be supported by all v9 components, while
+internal popper options can still belong to `PopperOptions` type in the `react-positioning` package.
+
+```tsx
+import { PositioningProps } from '@fluentui/react-positioning';
+
+export interface TooltipProps extends PositioningProps {}
+
+export interface MenuProps extends PositioningProps {}
+
+export interface PopoverProps extends PositioningProps {}
+```
+
+### Shorhand `placement` configuration
+
+The next step to this proposal is to make sure that all components which use a positioned element should allow users
+to configure popper through a `placement` prop that is behaves like a shorthand. The usage examples below will detail
+the usage of this shorthand.
+
+```tsx
+import { PositioningProps } from '@fluentui/react-positioning';
+
+export interface TooltipProps {
+  placement?: PositioningProps;
+}
+
+export interface MenuProps {
+  placement?: PositioningProps;
+}
+
+export interface PopoverProps {
+  placement?: PositioningProps;
+}
+```
+
+Example usage:
+
+```tsx
+// Shorthand for `position` and `align` options
```

Comment:
This isn't specified in this RFC, but since I mentioned it in my earlier comment, I wanted to amend my suggestion about the shorthand: It's probably best _not_ to have align-only shorthand (like ~~`placement="start"`~~), and just have either position-only like `placement="above"` (implies `align: 'center'`), or position-align like `placement="above-start"`. I.e.:
```ts
type PositioningShorthand =
  | 'above'
  | 'above-start'
  | 'above-end'
  | 'below'
  | 'below-start'
  | 'below-end'
  | 'before'
  | 'before-top'
  | 'before-bottom'
  | 'after'
  | 'after-top'
  | 'after-bottom';
```

If you still wanted to only specify align, you could still do it using the longhand syntax `placement={{ align: 'start' }}`, but that's probably not as common of a use-case.

Reply:
IMO this is implementation detail how we want to implement the shorthand values, but yes I agree, the values must have a specific order

Reply:
Added your helpful type enumeration